vcsim/main.go

```diff
@@ -110,12 +111,10 @@ func main() {
 	}
 
 	f := flag.Lookup("httptest.serve")
-	listen := f.Value.String()
-	if listen == "" {
-		listen = "127.0.0.1:8989"
-		_ = f.Value.Set(listen)
+	if f.Value.String() == "" {
+		_ = f.Value.Set(*listen) // propagate -listen unless -httptest.serve is specified
 	}
-	if err = updateHostTemplate(listen); err != nil {
+	if err = updateHostTemplate(*listen); err != nil {
```

Comment:
Is there any risk of a nill dereference?

Reply:
No, `listen` is returned by `flag.String` in this case.